gtest_parallel_tests.py

```diff
@@ -311,36 +311,44 @@ def test_times_worker():
   def test_log_file_names(self):
-    def root():
-      return 'C:\\' if sys.platform == 'win32' else '/'
-
-    self.assertEqual(
-      os.path.join('.', 'bin-Test_case-100.log'),
-      gtest_parallel.Task._logname('.', 'bin', 'Test.case', 100))
-
-    self.assertEqual(
-      os.path.join('..', 'a', 'b', 'bin-Test_case_2-1.log'),
-      gtest_parallel.Task._logname(os.path.join('..', 'a', 'b'),
-                                   os.path.join('..', 'bin'),
-                                   'Test.case/2', 1))
-
-    self.assertEqual(
-      os.path.join('..', 'a', 'b', 'bin-Test_case_2-5.log'),
-      gtest_parallel.Task._logname(os.path.join('..', 'a', 'b'),
-                                   os.path.join(root(), 'c', 'd', 'bin'),
-                                   'Test.case/2', 5))
-
-    self.assertEqual(
-      os.path.join(root(), 'a', 'b', 'bin-Instantiation_Test_case_2-3.log'),
-      gtest_parallel.Task._logname(os.path.join(root(), 'a', 'b'),
-                                   os.path.join('..', 'c', 'bin'),
-                                   'Instantiation/Test.case/2', 3))
-
-    self.assertEqual(
-      os.path.join(root(), 'a', 'b', 'bin-Test_case-1.log'),
-      gtest_parallel.Task._logname(os.path.join(root(), 'a', 'b'),
-                                   os.path.join(root(), 'c', 'd', 'bin'),
-                                   'Test.case', 1))
+    with tempfile.TemporaryDirectory() as root:
+      self.assertEqual(
+        os.path.join('.', 'bin-Test_case-100.log'),
+        gtest_parallel.Task._logname('.', 'bin', 'Test.case', 100))
+
+      self.assertEqual(
+        os.path.join('..', 'a', 'b', 'bin-Test_case_2-1.log'),
+        gtest_parallel.Task._logname(os.path.join('..', 'a', 'b'),
+                                    os.path.join('..', 'bin'),
+                                    'Test.case/2', 1))
+
+      self.assertEqual(
+        os.path.join('..', 'a', 'b', 'bin-Test_case_2-5.log'),
+        gtest_parallel.Task._logname(os.path.join('..', 'a', 'b'),
+                                    os.path.join(root, 'c', 'd', 'bin'),
+                                    'Test.case/2', 5))
+
+      self.assertEqual(
+        os.path.join(root, 'a', 'b', 'bin-Instantiation_Test_case_2-3.log'),
+        gtest_parallel.Task._logname(os.path.join(root, 'a', 'b'),
+                                    os.path.join('..', 'c', 'bin'),
+                                    'Instantiation/Test.case/2', 3))
+
+      self.assertEqual(
+        os.path.join(root, 'a', 'b', 'bin-Test_case-1.log'),
+        gtest_parallel.Task._logname(os.path.join(root, 'a', 'b'),
+                                    os.path.join(root, 'c', 'd', 'bin'),
+                                    'Test.case', 1))
+      MAX_PREFIX_LENGTH = 240
+      long_test_name = 'a' * (os.statvfs('.').f_namemax if hasattr(os,
+                                                                   'statvfs') else MAX_PREFIX_LENGTH + 1)
```

Comment:
Could you make this one a separate test? Maybe `test_truncates_file_name`

Also I think this test only makes sense on systems that have `statvfs`, can it run conditionally? This test requires OSes without `statvfs` to reject filenames that are longer which is maybe not what we want. I'm OK with not testing the MAX_PREFIX_LENGTH branch.